Applications/Utils/MeshEdit/removeMeshElements.cpp

```diff
@@ -11,35 +11,44 @@
  *              http://www.opengeosys.org/LICENSE.txt
  */
 
+#include <memory>
+
 // TCLAP
 #include "tclap/CmdLine.h"
 
-// ThirdParty/logog
-#include "logog/include/logog.hpp"
-
-// BaseLib
-#include "LogogSimpleFormatter.h"
+#include "Applications/ApplicationsLib/LogogSetup.h"
 
-// FileIO
-#include "Legacy/MeshIO.h"
-#include "readMeshFromFile.h"
+#include "FileIO/readMeshFromFile.h"
+#include "FileIO/writeMeshToFile.h"
 
-// MeshLib
-#include "Mesh.h"
+#include "MeshLib/Mesh.h"
 #include "MeshLib/Node.h"
-#include "Elements/Element.h"
-#include "MeshEnums.h"
-#include "MeshSearch/ElementSearch.h"
-#include "MeshEditing/RemoveMeshComponents.h"
+#include "MeshLib/Elements/Element.h"
+#include "MeshLib/MeshEnums.h"
+#include "MeshLib/MeshSearch/ElementSearch.h"
+#include "MeshLib/MeshEditing/RemoveMeshComponents.h"
+
+template <typename PROPERTY_TYPE>
+void searchByProperty(std::string const& property_name,
+                      std::vector<PROPERTY_TYPE> const& property_values,
+                      MeshLib::ElementSearch& searcher)
+{
+	for (auto property_value : property_values) {
+		const std::size_t n_removed_elements =
```

Comment:
Changed to `n_marked_elements` what does better refect the purpose of the variable.